Tools/CISupport/ews-build/steps.py

```diff
@@ -1720,23 +1720,29 @@ class CloseBug(buildstep.BuildStep, BugzillaMixin):
     flunkOnFailure = False
     haltOnFailure = False
 
-    def start(self):
-        self.bug_id = self.getProperty('bug_id', '')
-        if not self.bug_id:
-            self._addToLog('stdio', 'bug_id build property not found.\n')
-            self.descriptionDone = 'No bug id found'
-            self.finished(FAILURE)
-            return None
+    def __init__(self, **kwargs):
+        self.bug_id = ''
+        super(CloseBug, self).__init__(**kwargs)
 
+    def start(self):
+        self.bug_id = self.getProperty('bug_id', self.bug_id)
         rc = self.close_bug(self.bug_id)
         self.finished(rc)
         return None
 
     def getResultSummary(self):
+        if self.results == SKIPPED:
+            return {'step': 'No bug to close'}
         if self.results == SUCCESS:
             return {'step': 'Closed bug {}'.format(self.bug_id)}
         return {'step': 'Failed to close bug {}'.format(self.bug_id)}
```

Comment:
Actually, I missed noticing this default case handling (in this and previous few PRs). It doesn't look good. You should explicitly handle the cases which you want, and let buildbot default getResultSummary() handle the rest.

e.g.: https://github.com/WebKit/WebKit/blob/main/Tools/CISupport/ews-build/steps.py#L882 or https://github.com/WebKit/WebKit/blob/main/Tools/CISupport/ews-build/steps.py#L2834

Reply:
Thanks for the explanation here, will be updating these later today.